### ISSUE TITLE
docs/math.md preconditioner formulas contradict the implementation

### ISSUE THREAD

**Issue by sachncs** [sachncs/xaker#33]
## Problem

`docs/math.md` documents preconditioners as if they were a textbook series. Two of the four entries describe math the code does not implement:

### `Diagonal`

`docs/math.md:170-181` claims:

> "**`Diagonal`** … `build` extracts the per-row L1 norm of `K + lam I`: `d_i = 1/(|K_{i,:}| + epsilon)`"

But `xaker/solver/precond.py:99-115` actually computes:

```python
diag = torch.diagonal(kernel, dim1=-2, dim2=-1)          # diagonal entries only
out = torch.nn.functional.softplus(diag + lam.squeeze(-1).squeeze(-1))
out = out * self.scale.abs() + self.eps
```

That is a softplus-positive *diagonal* preconditioner (Jacobi on the diagonal), not an L1-norm row-sum inverse. The formula in the doc is the symmetric positive-definite Jacobi preconditioner — a different object.

### `Fast`

`docs/math.md:185-207` describes a generic low-rank-plus-diagonal preconditioner that draws `r` direction samples "from the kernel" and fits the rank-r factor as `U ∈ R^{n × r}`. The actual implementation in `xaker/solver/precond.py:166-183` uses a *fixed, learned* `lr_base` parameter (allocated once in `__init__`, not resampled per-build) and `lr_imp` per-build scaling:

```python
lr = self.lr_base[:, :length, :] * torch.nn.functional.softplus(self.lr_imp).unsqueeze(1)
```

So `Fast` is a learned low-rank preconditioner that has nothing to do with the angular sampling described in the math doc.

## Evidence

- `docs/math.md:170-181` ("Diagonal" L1-norm vs actual diagonal-softplus).
- `docs/math.md:185-207` ("Fast" generic resampling vs the fixed-parameter implementation).
- `xaker/solver/precond.py:99-115` (Diagonal.build) and `xaker/solver/precond.py:166-183` (Fast.build).
- `xaker/solver/precond.py:215-281` (Cccp.samples / Cccp.step use real angular sampling; this matches the math doc correctly — only Diagonal and Fast are misdescribed).

## Root Cause

The math doc was authored from the LAKER paper, then the implementation drifted (in particular, `Fast` is implemented as a learned parameter rather than a per-build random projection). The doc was not updated to match.

## Impact

- A reader who studies the math then reads the code cannot reconcile the two. The repo explicitly says "if a derivation in the paper contradicts the code, the code wins" (`docs/math.md:11`), but the *doc* itself contradicts the code.
- The codebase has a custom grader (`xaker/rubric/grader.py`) that scores `correctness: 3/3` based on file presence; it does not detect math-doc drift.
- Reviewers of the upcoming paper will find the math doc inconsistent with the released code.

## Recommendation

Rewrite `docs/math.md` sections 5 ("Preconditioner parameterisations") to match the implementation:

- **Diagonal**: softmax-positived diagonal scaling with a learned `scale`. Make the Jacobi-algebra explicit.
- **Fast**: learned fixed basis `lr_base ∈ R^{H × 2048 × r}` and per-step `softplus(lr_imp)` multiplier.
- **Cccp**: angular sampling + Tyler's M-estimator fixed point + eigh. (No change; this section is correct.)
- **Identity**: trivial. (No change.)

Add a one-line callout at the top: "The implementation is the spec; if a derivation here contradicts `xaker/solver/precond.py`, the code wins."

## Acceptance Criteria

- [ ] `Diagonal` math section matches `xaker/solver/precond.py:99-115` byte-for-byte formula.
- [ ] `Fast` math section matches `xaker/solver/precond.py:166-183`.
- [ ] The `correctness` rubric dimension in `xaker/rubric/grader.py` is taught to *optionally* verify math-doc drift (check that `docs/math.md` references `softplus` when describing Diagonal).
- [ ] Math doc passes a manual review against the actual source.

## Verification

```bash
git grep -n "softplus" xaker/solver/precond.py
git grep -n "per-row L1 norm" docs/math.md
```

The first must return matches; the second must not.

## Dependencies

None.

## Risk

None. Documentation only.
